src/lib/components/home/artwork_rails/artwork_rail.tsx

```diff
@@ -207,13 +207,21 @@ class ArtworkRail extends React.Component<Props & RelayPropsWorkaround, State> {
     if (this.state.loadFailed) {
       return null
     }
+    const hasFetched = this.props.relay && this.props.relay.variables.fetchContent
 
     const sideMargin = isPad ? 40 : 20
     const style: any = { marginLeft: sideMargin, marginRight: sideMargin }
-    if (!(this.props.relay && this.props.relay.variables.fetchContent)) {
+    if (!hasFetched) {
       // if there's no content, set minHeight to prevent spinners from all the rails showing
       style.minHeight = 400
     }
+
+    const hasArtworks = hasFetched && this.props.rail.results && this.props.rail.results.length
+    if (!hasArtworks) {
```

Comment:
@sarahscott You could rewrite the check that way so it'll only check if the results array is empty after it's done fetching data.

```javascript
    if (hasFetched) {
      const hasArtworks = this.props.rail.results && this.props.rail.results.length
      if (!hasArtworks) {
        // if the data has been fetched but there are no results, hide the whole thing
        return null
      }
    } else {
      // if there's no content, set minHeight to prevent spinners from all the rails showing
      style.minHeight = 400
    }

    return (
      <View accessibilityLabel="Artwork Rail" style={{ paddingBottom: this.state.expanded ? 0 : 12 }}>
        <Header rail={this.props.rail} handleViewAll={this.handleViewAll}/>
        <View style={style}>
          { this.renderModuleResults() }
        </View>
      </View>
    )
```